src/Microsoft.AspNet.OData.Shared/EnableQueryAttribute.cs

```diff
@@ -611,7 +611,7 @@ public virtual object ApplyQuery(object entity, ODataQueryOptions queryOptions)
                 {
                     IQueryable queryable = singleResultCollection as IQueryable;
                     queryable = ApplyQuery(queryable, queryOptions);
-                    return SingleOrDefault(queryable, actionDescriptor);
+                    return QueryHelpers.SingleOrDefault(queryable, actionDescriptor);
```

Comment:
Is this change necessary? Having the QueryHelpers class? The implementation of SingleOrDefault in this class and what was there before is similar.

Reply:
I moved the method to a helper class (`QueryHelpers`) because now it's also used by the `EnableNestedPathsAttribute` class. So instead of calling `EnableQueryAttribute.SingleOrDefault` from `EnableNestedPathsAttribute`, they now both call `QueryHelpers.SingleOrDefault`.